Readme.md

```diff
@@ -386,6 +388,174 @@ export function MyForm({ isNewItem, addItem, updateItem }: Props): JSX.Element =
+  );
+}
+```
+
+Validations also work with nested objects
+
+```ts
+import { useForm } from 'react-use-form-library';
+
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      address: {
+        streetName: '',
+      },
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+    validations: {
+      address: {
+        streetName: 'required',
+      },
+    }
+  });
+
+  // ...
+```
+
+Since the fields property within FormField is also based on a proxy object, you can access nested properties without
+explicitely declaring them
+
+```ts
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {},
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        value={fields.bankDetails.fields?.address.fields?.streetName.value}
+        onChange={(v) =>
+          fields.bankDetails.fields?.address.fields?.streetName.onChange(
+            v.target.value
+          )
+        }
+      />
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+---
+
+## FieldSet
+
+A FieldSet is basically an array of FormFields. Contrary to nested objects, a FieldSet will only be instantiated if it is **explicitely** declared in the model.
+
+This is because at the moment of field instantiation, the addField method within a Form (or a FormField if it is a nested object) will as if the model value is actually an Array. It is not possible of implicitely determine if the field is an array based only on the type.
+
+The FieldSet class extends Array, so you can access all methods an array has, plus custom methods that work similarly as those in FormField (more in the [API](#fieldset-1))
+
+```ts
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      emails: ['example@email.com'],
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      // a FieldSet is iterable
+      {fields.emails.map((field, index) => {
+        // Every FieldSet item is a FormField
+        return (
+          <div className="input">
+            <input
+              value={field.value}
+              onChange={(v) => field.onChange(v.target.value)}
+            />
+            // FieldSet has a helpful removeField method
+            <button onClick={() => fields.emails.remove(index)}>remove</button>
+          </div>
+        );
+      })}
+      // FieldSet also has an addFields method to create new items
+      <button onClick={() => fields.emails.addFields('')}>Add Field</button>
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+### Adding Items
+
+The `addFields` methods takes an array of items that can be either a simple value or an object with value and validation. Being an array extension, it is also possible to add items to FieldSet using array methods like `push`;
```

Comment:
I'm not sure if you're doing yourself a favor by extending `Array`. While in theory it makes sense, I haven't yet witnessed a single case where it turned out to be a good idea in the end.

Reply:
If instead you return a `fields`-Array from the property, it would also be closer to the API of nested objects:
```ts
fields.emails.fields.each
fields.emails.add('newmail')
```